pkg/kubelet/cm/dra/manager.go

```diff
@@ -126,39 +114,28 @@ func (m *ManagerImpl) PrepareResources(pod *v1.Pod) error {
 
 				klog.V(3).InfoS("NodePrepareResource succeeded", "response", response)
 
-				// NOTE: Passing CDI device names as annotations is a temporary solution
-				// It will be removed after all runtimes are updated
-				// to get CDI device names from the ContainerConfig.CDIDevices field
-				annotations, err := generateCDIAnnotations(resourceClaim.UID, driverName, response.CdiDevices)
-				if err != nil {
-					return fmt.Errorf("failed to generate container annotations, err: %+v", err)
-				}
-
-				// Cache prepared resource
-				err = m.cache.add(
+				// @TODO(moshe010): We are adding claimInfo to cache and sync to checkpoint after NodePrepareResource call.
+				// This has edge case when kubelet is restarted between NodePrepareResource and syncToCheckpoint. This may
+				// result not calling NodeUnprepareResource because the resource claim is not the cache. We should find
+				// better solution before moving to Beta.
```

Comment:
```suggestion
				// TODO: We are adding the claimInfo struct to the cache and syncing it to the checkpoint *after* the NodePrepareResource call has completed.
				// This will cause issues if the kubelet gets restarted between NodePrepareResource and syncToCheckpoint. It will
				// result in not calling NodeUnprepareResource for this claim because not claimInfo will be synced back to the cache for it after the restart. We need to resolve this issue before moving to beta.
```